Added crud api calls for a logged in user w.r.t favorites

vasdhara opened the pull request.

vasdhara committed:
Added crud api calls for a logged in user w.r.t favorites

jeremy-loss commented on src/requests.ts:
```diff
@@ -34,16 +35,20 @@ export function authenticatedRequest(
   const request: IFetchExecutor = fetchExecutor || fetch
   const url = baseURL(environment) + uri
   /* tslint:disable:no-console */
   console.log(`[MHBO-JS] ${method} | ${url}`)
```
Can we clean up these console logs

jeremy-loss commented on src/users/__tests__/deleteFavorite.test.ts:
```diff
@@ -0,0 +1,45 @@
+    expect(mockFetch.mock.calls.length).toBe(1)
+  })
+
+  it("should call the search endpoint with correct data in body", () => {
```
Not super important, but can we  just update the description from the original template of "search endpoint" and make it specific to delete.

jeremy-loss commented on src/users/index.ts:
```diff
@@ -0,0 +1,65 @@
+      body
+    )
+  },
+  deleteFavorite: (id: number) => {
```
I added the ability for administrators to update another user's data and checked that they are authorized when the call is made. Can we make the user_id a parameter that is passed in instead of pulled from the token. The React client will do this decoding of the token and pass it to the API. When we make an OpenAPI definition for the API, generated code will create a method with 2 parameters anyway, so it will be easier to follow along across programming languages.

What do you think about having a method in mhbo-js that clients use to extract the user_id? So instead of the React app having to know specifically that the field name is "user_id" in the token, you can just call getUserId(jsonwebtoken). That way if we change the field or modify defaults we can do it in one place. We could just place it in users.getUserId or maybe put it in a Token utility of some sort. What do you think?

vasdhara replied:
Added a token utility to get userId along with IToken type so that other information can be extracted from the token later.

jeremy-loss commented on src/users/index.ts:
```diff
@@ -0,0 +1,65 @@
+import jsonwebtoken from "jsonwebtoken"
+import { requestGet, requestPost, requestDelete } from "../entityRequest"
```
Lint, which is run on publish,  is complaining about the order of this being alphabetized- requestDelete should be first.

jeremy-loss commented on src/users/index.ts:
```diff
@@ -0,0 +1,65 @@
+      fetchExecutor
+    )
+  },
+  getFavorites: () => {
```
Can you add user_id as a parameter so admins can lookup other user's favorites.

jeremy-loss reviewed (changes requested):
This looks great! Just some small changes requested in the comments. Since you are the only one update this API at the moment, can you increment the version in package.json to 1.0.24.

vasdhara committed:
updated package number
fixed lint errors
removed console logs
added token utility and types
update favorites method calls to take userId in params
updated template descriptions of all tests
